### ISSUE TITLE
Cannot create docker image using `DockerImage`

### ISSUE THREAD

**Issue by bloomdt-uw** [nrdg/cloudknot#264]
`docker-image vis-rois` config must already exist in order to create `DockerImage` object to reuse in Knot.

```
di = ck.DockerImage(
    name="vis-rois",
    func=vis_rois,
    base_image="python:3.8",
    github_installs=["https://github.com/bloomdt-uw/pyAFQ.git@enh-565-subsample-streamlines"],
    overwrite=True
)

ResourceDoesNotExistException: Could not find docker-image vis-rois in config_file ~/.aws/cloudknot
dockerimage.py:166
```

**Comment by bloomdt-uw** [comment]
@richford We discussed last week; didn't see it logged so may be duplicate.

**Comment by richford** [comment]
@bloomdt-uw, This should be fixed with the merge of #265. I modified the unit tests as well but would you mind testing this out from the latest master branch using your real-world example above?

**Comment by bloomdt-uw** [comment]
Copying from slack:

> Are the cloudknot definitions not backward compatible?

> To test fix -- I uninstalled the released version, then installed from github. Now getting KeyError:

Code:
```
di = ck.DockerImage(
    name="vis-rois",
    func=vis_rois,
    base_image="python:3.8",
    github_installs=["https://github.com/bloomdt-uw/pyAFQ.git@enh-565-subsample-streamlines"],
    overwrite=True
)
```

Output:
```
~/anaconda/envs/pyAFQ/lib/python3.8/site-packages/cloudknot/dockerimage.py in __init__(self, name, func, script_path, dir_name, base_image, github_installs, username, overwrite)
    183                 params_changed = True
    184             else:
--> 185                 self._region = config.get(section_name, "region")
    186                 self._profile = config.get(section_name, "profile")
    187                 self.check_profile_and_region()
~/anaconda/envs/pyAFQ/lib/python3.8/configparser.py in get(self, section, option, raw, vars, fallback)
    790         except KeyError:
    791             if fallback is _UNSET:
--> 792                 raise NoOptionError(option, section)
    793             else:
    794                 return fallback
NoOptionError: No option 'region' in section: 'docker-image vis-rois'
```

cloudknot config:
```
[docker-image vis-rois]
function-hash = 8771819327362
build-path = /Users/bloomdt/NRDG/vizpathway/cloudknot_docker_vis-rois_nt0cgtkf
script-path = /Users/bloomdt/NRDG/vizpathway/cloudknot_docker_vis-rois_nt0cgtkf/
vis-rois.py
docker-path = /Users/bloomdt/NRDG/vizpathway/cloudknot_docker_vis-rois_nt0cgtkf/
Dockerfile
req-path = /Users/bloomdt/NRDG/vizpathway/cloudknot_docker_vis-rois_nt0cgtkf/req
uirements.txt
base-image = python:3.8
github-imports = https://github.com/bloomdt-uw/pyAFQ.git@enh-565-subsample-strea
mlines
username = cloudknot-user
images = cloudknot/vis-rois:vis_rois-2020-10-22T12-35-55
repo-uri = 454929164628.dkr.ecr.us-west-2.amazonaws.com/cloudknot:vis_rois-2020-10-22T12-35-55
clobber-script = True
```
